script.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each(array)
+    index = 0
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  def my_each_with_index(array)
+    index = 0
+    while index < array.length
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
+
+  def my_select(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(array)
+    p my_select(array) { |x| yield(x) }.length == array.length
+  end
+
+  def my_any?(array)
+    p my_select(array) { |x| yield(x) }.length.positive? ? true : false
+  end
+
+  def my_none?(array)
+    p my_select(array) { |x| yield(x) }.length.length.zero? ? true : false
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?
> Remember to test it not only for arrays but for ranges as well
- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false, true, []].my_none? == [nil, false, true, []].none? => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77690309-b60feb80-6f68-11ea-8424-8fc0e6b3fc59.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690365-cd4ed900-6f68-11ea-9933-f73b0091208d.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77690513-05eeb280-6f69-11ea-8535-9343ab8052ea.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77690678-4d753e80-6f69-11ea-992d-1b0db59e2b3c.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77690884-9c22d880-6f69-11ea-9bbd-f0eddc09c7e2.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg , &block)
  end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?